### PR TITLE
[EventDispatcher] TraceableEventDispatcher resets event listener priorities

### PR DESCRIPTION
**Pull request by c960657**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #15550 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/EventDispatcher/Tests/Debug/TraceableEventDispatcherTest.php [hunk 1]
```diff
@@ -138,12 +155,12 @@ public function testDispatchCallListeners()
 
         $dispatcher = new EventDispatcher();
         $tdispatcher = new TraceableEventDispatcher($dispatcher, new Stopwatch());
-        $tdispatcher->addListener('foo', $listener1 = function () use (&$called) { $called[] = 'foo1'; });
-        $tdispatcher->addListener('foo', $listener2 = function () use (&$called) { $called[] = 'foo2'; });
+        $tdispatcher->addListener('foo', $listener1 = function () use (&$called) { $called[] = 'foo1'; }, 10);
+        $tdispatcher->addListener('foo', $listener2 = function () use (&$called) { $called[] = 'foo2'; }, 20);
```
Comment (stof) [inline src/Symfony/Component/EventDispatcher/Tests/Debug/TraceableEventDispatcherTest.php]: the ``$listener1`` and ``$listener2`` variables are actually unused

**Comment by stof** [comment]
@c960657 what is the lower Symfony version where this patch can be applied ? The bug seems to impact 2.7+

File: src/Symfony/Component/EventDispatcher/Tests/Debug/TraceableEventDispatcherTest.php (hunk 1 above)
Comment (c960657) [inline src/Symfony/Component/EventDispatcher/Tests/Debug/TraceableEventDispatcherTest.php]: Fixed

**Comment by c960657** [comment]
@stof The patch applies cleanly to 2.7+. The bug seems to have always been there, even back in 2.3 when TraceableEventDispatcher belonged to HttpKernel.

**Comment by fabpot** [comment]
@c960657 If the bug is present in 2.3, we need to fix it there first. Can you work on creating a PR for 2.3?

**Comment by c960657** [comment]
I have made separate pull requests for the 2.3 (#17482) and 2.7 (#17483) branches.

The patch for master applies cleanly to the 2.7 branch (i.e. git does not complain), but it would break BC, because EventDispatcherInterface::getListenerPriority() is not defined prior to 3.0.

**Comment by fabpot** [comment]
@c960657 Sorry for that but I've closed the other PRs as the listener priority is only available as of 2.8, so I'm going to merge this one on 2.8.

**Comment by fabpot** [comment]
Thank you @c960657.

**Pull request closed without merging**